sympde/topology/base_analytic_mapping.py

```diff
@@ -0,0 +1,223 @@
+            obj._inv_jac_eval = lambdify_sympde( obj._logical_coordinates, obj._inv_jac)
+            obj._metric_eval = lambdify_sympde( obj._logical_coordinates, obj._metric)
+            obj._metric_det_eval = lambdify_sympde( obj._logical_coordinates, obj._metric_det)
+        else:
+            raise TypeError("BaseAnalyticMapping should have an expression")
+        
+        return obj
+
+    
+    #--------------------------------------------------------------------------
+    #Abstract Interface : 
+    
+    
+    def _evaluate_domain( self, domain ):
+        assert(isinstance(domain, BasicDomain))
+        return MappedDomain(self, domain)
+    
+    def _evaluate( self, *Xs ):
+        #int, float or numpy arrays 
+        if self._func_eval is None :
+            raise TypeError("not a callable object")
+        else :
+            assert len(Xs)==self.ldim
+            Xshape = np.shape(Xs[0]) 
+            for X in Xs:
+                assert np.shape(X) == Xshape
+            return tuple( f( *Xs ) for f in self._func_eval)
+        
+    def _jacobian_evaluate( self, *Xs ):
+        #int, float or numpy arrays 
+        if self._jac_eval is None:
+            raise TypeError("not a callable object")
+        else :
+            assert len(Xs)==self.ldim
+            Xshape = np.shape(Xs[0]) 
+            for X in Xs:
+                assert np.shape(X) == Xshape
+            return self._jac_eval( *Xs ) 
+        
+    def _jacobian_inv_evaluate( self, *Xs ):
+        #int, float or numpy arrays 
+        if self._inv_jac_eval is None:
+            raise TypeError("not a callable object")
+        else :
+            assert len(Xs)==self.ldim
+            Xshape = np.shape(Xs[0]) 
+            for X in Xs:
+                assert np.shape(X) == Xshape
+            return  self._inv_jac_eval( *Xs ) 
+        
+    def _metric_evaluate( self, *Xs ):
+        if self._metric_eval is None:
+            raise TypeError("not a callable object")
+        else :
+            assert len(Xs)==self.ldim
+            Xshape = np.shape(Xs[0]) 
+            for X in Xs:
+                assert np.shape(X) == Xshape
+            return  self._metric_eval( *Xs ) 
+        
+    def _metric_det_evaluate( self, *Xs ):
+        if self._metric_det_eval is None:
+            raise TypeError("not a callable object")
+        else :
+            assert len(Xs)==self.ldim
+            Xshape = np.shape(Xs[0]) 
+            for X in Xs:
+                assert np.shape(X) == Xshape
+            return self._metric_det_eval( *Xs ) 
+    
+    def __call__( self, *args ):
+        if len(args) == 1 and isinstance(args[0], BasicDomain):
+            return self._evaluate_domain(args[0])
+        elif all(isinstance(arg, (int, float, Symbol, np.ndarray)) for arg in args):
+            return self._evaluate(*args)
+        else:
+            raise TypeError("Invalid arguments for __call__")
+        
+        
+    def jacobian_eval( self, *args ):
```

Comment:
The use of two consecutive blank lines should be avoided inside a class definition.
```suggestion
            raise TypeError("Invalid arguments for __call__")

    def jacobian_eval( self, *args ):
```